### PR TITLE
tweak checkify docs.

### PR DESCRIPTION
**Pull request by LenaMartens**
Rewrote the sections to be a bit less verbose.

Added some examples of checkify under various transforms.

### CODE REVIEW

File: docs/debugging/checkify_guide.md [hunk 1]
```diff
@@ -145,14 +149,119 @@ err.throw()
...
 The API for selecting which automatic checks to enable is based on Sets. See {mod}`jax.experimental.checkify` for more details.
 
+## `checkify` under JAX transformations.
+
+As demonstrated in the examples above, a checkified function can be happily
+jitted. Here's a few more examples of `checkify` with other  JAX
+transformations. Note that checkified functions are functionally pure, and
+should trivially compose with all JAX transformations!
+
+### vmap/pmap
```
Comment (sharadmv) [inline docs/debugging/checkify_guide.md:161]: Not: wrap in back ticks

File: docs/debugging/checkify_guide.md [hunk 2]
```diff
@@ -145,14 +149,119 @@ err.throw()
...
+  checkify.check(i >= 0, "index needs to be non-negative!")
+  return x[i]
+
+checked_f = checkify.checkify(f, errors=checkify.all_errors)
+err, out = checked_f(jnp.ones((3, 5)), jnp.array([-1, 2, 100]))
+err.throw()
+# ValueError: index needs to be non-negative! (check failed at <...>:2 (f))
+```
+
+### pjit
```
Comment (sharadmv) [inline docs/debugging/checkify_guide.md:195]: Nit: backticks

File: docs/debugging/checkify_guide.md [hunk 3]
```diff
@@ -145,14 +149,119 @@ err.throw()
...
+  in_axis_resources=PartitionSpec('x', None),
+  out_axis_resources=(None, PartitionSpec('x', None)))
+ 
+with maps.Mesh(mesh.devices, mesh.axis_names):
+ err, data = f(input_data)
+err.throw()
+# ValueError: divided by zero at <...>:4 (f)
+```
+
+### grad
```
Comment (sharadmv) [inline docs/debugging/checkify_guide.md:216]: Same here

**Review by sharadmv — APPROVED** [review]
Thanks!!

**Comment by LenaMartens** [comment]
Thanks for the review! Applied your suggestions.

**Pull request merged**